### PR TITLE
Add Registered Model Alias to the Model Registry Doc Section

### PR DESCRIPTION
**Pull request by arpitjasa-db**
## Related Issues/PRs

https://github.com/mlflow/mlflow/pull/8001
https://github.com/mlflow/mlflow/pull/8055
https://github.com/mlflow/mlflow/pull/8094
https://github.com/mlflow/mlflow/pull/8164
https://github.com/mlflow/mlflow/pull/8377

## What changes are proposed in this pull request?

We are updating the Model Registry Doc section to include aliases and provide an example workflow. We also fix a alias name length to match the UC Registry limit.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Screenshot 2023-05-16 at 1 36 11 PM
Screenshot 2023-05-16 at 1 36 30 PM
Screenshot 2023-05-16 at 1 36 43 PM

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

We are updating the Model Registry Doc section to include aliases and provide an example workflow. We also fix a alias name length to match the UC Registry limit.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Review by harupy — APPROVED** [review]
Looks good to me!

File: docs/source/model-registry.rst [hunk 1]
```diff
@@ -600,6 +605,72 @@ save, log, register, and load from the Model Registry and score.
     <Yay!! Another good phone interview. I nailed it!!> -- {'neg': 0.0, 'neu': 0.446, 'pos': 0.554, 'compound': 0.816}
     <This is INSANE! I can't believe it. How could you do such a horrible thing?> -- {'neg': 0.357, 'neu': 0.643, 'pos': 0.0, 'compound': -0.8034}
 
+Using Registered Model Aliases
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+    Registered model aliases are experimental and subject to change.
```
Comment (smurching) [inline docs/source/model-registry.rst:611]: Did we forget to annotate the actual `set_registered_model_alias` API (https://mlflow.org/docs/latest/python_api/mlflow.client.html?highlight=set_registered_model_alias#mlflow.client.MlflowClient.set_registered_model_alias) as experimental when we first released it? I don't think we can retroactively mark as experimental

File: mlflow/utils/validation.py [hunk 2]
```diff
@@ -51,7 +51,7 @@
 MAX_MODEL_REGISTRY_TAG_KEY_LENGTH = 250
 MAX_MODEL_REGISTRY_TAG_VALUE_LENGTH = 5000
 MAX_EXPERIMENTS_LISTED_PER_PAGE = 50000
-MAX_REGISTERED_MODEL_ALIAS_LENGTH = 256
+MAX_REGISTERED_MODEL_ALIAS_LENGTH = 255
```
Comment (smurching) [inline mlflow/utils/validation.py:54]: Is this a breaking change for OSS?

File: docs/source/model-registry.rst [hunk 3]
```diff
@@ -600,6 +605,72 @@ save, log, register, and load from the Model Registry and score.
...
+    from sklearn.ensemble import RandomForestRegressor
+
+    import mlflow
+    import mlflow.sklearn
+
+    with mlflow.start_run(run_name="YOUR_RUN_NAME") as run:
+        params = {"n_estimators": 5, "random_state": 42}
+        sk_learn_rfr = RandomForestRegressor(**params)
+
+        # Log parameters and metrics using the MLflow APIs
```
Comment (smurching) [inline docs/source/model-registry.rst:638]: I'd recommend removing some of the extraneous `log_params` etc to keep this doc section focused on aliases, especially given the length of the doc page already

**Review by smurching — COMMENTED** [review]
Thanks for the PR @arpitjasa-db ! Left some initial comments

File: docs/source/model-registry.rst (hunk 1 above)
Comment (arpitjasa-db) [reply docs/source/model-registry.rst:611]: Yeah 🤦‍♂️ I didn't realize that the API docs were auto-generated from the function comments. Do you think it makes more sense to remove `experimental` from here or go add it to the API docs?

File: mlflow/utils/validation.py (hunk 2 above)
Comment (arpitjasa-db) [reply mlflow/utils/validation.py:54]: Yeah marked it on the PR

File: docs/source/model-registry.rst (hunk 1 above)
Comment (smurching) [reply docs/source/model-registry.rst:611]: I think we can just remove it from here, we don't expect to drop/deprecate the feature anyways or make breaking API changes

File: mlflow/utils/validation.py (hunk 2 above)
Comment (smurching) [reply mlflow/utils/validation.py:54]: Got it, I'm ok with this given aliases just shipped, but we should double check with @dbczumar. It doesn't seem too terrible if the SqlAlchemyStore etc support one extra char in alias names (allows 256 char alias names) than UC does (only allows 255 char alias names) long term, if it actually comes up it's probably not too hard to support the extra char in UC

Comment (smurching) [reply mlflow/utils/validation.py:54]: Got it, I'm ok with this given aliases just shipped, but we should double check with @dbczumar. It doesn't seem too terrible if the SqlAlchemyStore etc support one extra char in alias names (allows 256 char alias names) than UC does (only allows 255 char alias names) long term, if it actually comes up it's probably not too hard to support the extra char in UC

**Review by smurching — APPROVED** [review]
LGTM

**Pull request merged**